### PR TITLE
Add support to hide metadata in log line

### PR DESCRIPTION
**Pull request by timlohse1104**
See https://github.com/pinojs/pino-pretty/issues/444
---

With option `--hideMetadata` it is possible to transform a basic log line like this:

`[25.7.2023, 15:57:32] INFO: [Nest] 25.7.2023, 15:57:32 [RoutesResolver] | OcrController {/ocr}: {"context":"RoutesResolver"}`

into a simulated NestJS log line like this (provided you use the `messageFormat` option):

`[Nest] 25.7.2023, 15:57:32 [RoutesResolver] | OcrController {/ocr}: {"context":"RoutesResolver"}`

`--hideMetadata` option effectively hides `epoch`, `level`and other metadata like `reqId` from beeing printed in front of the actual log message.

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

File: index.js [hunk 1]
```diff
@@ -127,39 +128,42 @@ function prettyFactory (options) {
...
-    if (opts.levelFirst && prettifiedLevel) {
-      line = `${prettifiedLevel}`
-    }
 
-    if (prettifiedTime && line === '') {
-      line = `${prettifiedTime}`
-    } else if (prettifiedTime) {
-      line = `${line} ${prettifiedTime}`
-    }
+    if (!opts.hideMetadata) {
```
Comment (jsumners) [inline index.js:133]: Can this be done in some other way than continually nesting the core code of this function further and further to the right?

**Review by jsumners — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (timlohse1104) [reply index.js:133]: Do you have something in mind. I could see some options. One with higher refactoring effort like using strategy pattern (https://refactoring.guru/design-patterns/strategy) and some with lower effort like using switch statements/  lookup tables.

Comment (jsumners) [reply index.js:133]: I think a simple approach would be to move the now optional code path into a function.

Comment (timlohse1104) [reply index.js:133]: Added an approach

File: index.js [hunk 2]
```diff
@@ -206,6 +178,46 @@ function prettyFactory (options) {
     }
 
     return line
+
+    function createLineMetadata () {
```
Comment (jsumners) [inline index.js:182]: Does this function need to be defined as an inner function of `prettyFactory`? Or can we put it in `lib/utils`?

Comment (timlohse1104) [reply index.js:182]: We could put it in `lib/utils` but would have to pass many parameters like `log`, `colorizer`, `levelKey` etc. Think i would prefer to leave it in index.js. What do you think about the naming?

Comment (jsumners) [reply index.js:182]: My concern is legibility and maintainability. The inlined function is easier to read than the nested conditional version, but it is still a lot of code deeply nested within another function.

I think the function name is fine.

Comment (jsumners) [reply index.js:182]: @timlohse1104 I think you'll find it much easier to accommodate the ask now that #453 has been merged. Unfortunately, it'll be easier to start with a fresh PR. Rebasing this one may be too difficult.

If you have any questions, don't hesitate to ask.

**Pull request closed without merging**